tests/tr1/include/temp_file_name.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <random>
+#include <string>
+
+[[nodiscard]] inline std::string temp_file_name() {
+    std::string ret{"temp_file_"};
+
+    std::random_device rd;
+
+    for (int i = 0; i < 64; ++i) { // 64 hexits = 256 bits of entropy
+        ret.push_back("0123456789ABCDEF"[rd() % 16]);

```

Comment:
Why not use `uniform_int_distribution` instead of chopping the seeds from random device?

Reply:
I was a lazy kitty, rolling around in sunbeams and taking advantage of my knowledge of the input and output ranges. (As `random_device`'s range is 2<sup>32</sup> and I want 2<sup>4</sup>, modulo is safe here.)

As modulo is *not* safe in general ([citation](https://channel9.msdn.com/Events/GoingNative/2013/rand-Considered-Harmful)), and it's possible that this code could be modified (unlikely) or copied/imitated elsewhere and then modified for a different engine or desired range (more likely; note that `[a-z0-9]` or `[a-zA-Z0-9]` wouldn't be powers of 2), using `uniform_int_distribution` is the right call, thanks. I'll make the change.